docs/pubsub-usage.rst

```diff
@@ -272,3 +272,30 @@ Fetch messages for a pull subscription without blocking (none pending):
    >>> messages = [recv[1] for recv in received]
    >>> [message.message_id for message in messages]
    []
+
+Fetch pending messages, acknowledging those whose processing doesn't raise an
+error:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> client = pubsub.Client()
+   >>> topic = client.topic('topic_name')
+   >>> subscription = topic.subscription('subscription_name')
+   >>> with topic.batch() as batch:
+   ...     batch.publish('this is the first message_payload')
+   ...     batch.publish('this is the second message_payload',
+   ...                   attr1='value1', attr2='value2')
+   >>> from gcloud.pubsub.subscription import AutoAck
+   >>> with AutoAck(subscription, max_messages=10) as ack:
+   ...     for ack_id, message in ack.items():
+   ...        try:
+   ...            do_something_with(message)
+   ...        except:
+   ...            del ack[ack_id]
```

Comment:
Thanks for the update. I still feel a little iffy on it (it feels brittle / easy for a user to "do the wrong thing") but I'm not going to stop you.

As I mentioned above, this sample still won't work in Python 3 since it mutates the dictionary while using a dictionary items iterator.